tests/unit/package_managers/test_general.py

```diff
@@ -132,3 +144,80 @@ def test_extract_git_info(url: str, nonstandard_info: Any) -> None:
     }
     info.update(nonstandard_info or {})
     assert general.extract_git_info(url) == info
+
+
+@pytest.mark.asyncio
+async def test_async_download_binary_file(tmp_path: Path) -> None:
+    url = "http://example.com/file.tar"
+    download_path = tmp_path / "file.tar"
+    chunk_size = 8192
+
+    # Create mock response and session
+    response, session = MagicMock(), MagicMock()
+    response.content.read = MockReadChunk().read_chunk
+
+    async def mock_aenter() -> MagicMock:
+        return response
+
+    session.get().__aenter__.side_effect = mock_aenter
+
+    await _async_download_binary_file(session, url, download_path, chunk_size=chunk_size)
+
+    # Verify that the file was downloaded correctly
+    with open(download_path, "rb") as f:
+        assert f.read() == b"first_chunk-second_chunk-"
+
+    # # Verify that session.get was called with the correct arguments
+    assert session.get.called
+    assert session.get.call_args == mock.call(url, auth=None, raise_for_status=True)
+
+
+@pytest.mark.asyncio
+@mock.patch("aiohttp.ClientSession")
+@mock.patch("aiohttp_retry.RetryClient")
+@mock.patch("cachi2.core.package_managers.general._async_download_binary_file")
+async def test_async_download_files(
+    mock_download_file: MagicMock,
+    mock_retry_client: MagicMock,
+    mock_session: MagicMock,
+    tmp_path: Path,
+    mock_async_download_binary_file: MagicMock,
+) -> None:
+    files_to_download: Dict[str, Union[str, PathLike[str]]] = {
+        "file1": str(tmp_path / "path1"),
+        "file2": str(tmp_path / "path2"),
+        "file3": str(tmp_path / "path3"),
+    }
+
+    concurrency_limit = 2
+
+    # Set up the mock for _async_download_binary_file
+    mock_download_file.return_value = mock_async_download_binary_file
+
+    await async_download_files(files_to_download, concurrency_limit)
+
+    # Assert that mock_download_file was called for each file
+    assert mock_download_file.call_count == 3
+
+    # Assert that mock_download_file was called with the correct arguments
+    for call in mock_download_file.mock_calls:
+        # call looks like this:
+        # call(<AsyncMock name='RetryClient().__aenter__()' id='1'>, 'file1', '/tmp/path1')
+        file, path = re.findall(r"'([^']+)'", str(call))[-2:]
+        assert file, path in files_to_download.items()
+
+
+@pytest.mark.asyncio
+@mock.patch("cachi2.core.package_managers.general.log")
+async def test_on_request_start_logs_debug(mock_log: MagicMock) -> None:
```

Comment:
Minor detail, but we could also test the scenario in which `current_attempt <= 1`